### PR TITLE
[Args] adds `optional_value_of` to `HasParsedArgs`, updated README

### PR DESCRIPTION
**Pull request by philgebhardt**
Creates convenience functions `optional_value_of` and `validated_optional_value_of`.

### CODE REVIEW

File: src/tst/mod.rs [hunk 1]
```diff
@@ -356,6 +356,121 @@ mod validated_value_of {
     }
 }
 
+mod optional_value_of {
```
Comment (philgebhardt) [inline src/tst/mod.rs:359]: Really like the way you organized these tests BTW.

File: src/lib.rs [hunk 2]
```diff
@@ -288,6 +288,35 @@ impl Args {
...
+        }
+    }
+
+    /// Retrieves the optional value of the `Opt` identified by `opt_name`, casts it to
+    /// the type specified by `T`, runs all provided `Validation`s, and wraps it in an Option<T>.
+    ///
+    /// # Failures
+    ///
+    /// See `validated_value_of`
+    fn validated_optional_value_of<T: FromStr>(&self, opt_name: &str, validations:
```
Comment (mattforni) [inline src/lib.rs]: Would prefer the names reversed (i.e. `optional_validated_value_of`). Just makes more sense in my mind that way. It's optional, and if present, is validated. I know it's nitpicky

File: src/tst/mod.rs (hunk 1 above)
Comment (mattforni) [reply src/tst/mod.rs:359]: Thank you

File: src/tst/mod.rs [hunk 3]
```diff
@@ -356,6 +356,121 @@ mod validated_value_of {
...
+                assert!(result.is_ok());
+                let optional = result.unwrap();
+                assert!(optional.is_some());
+                assert_eq!(0i32, optional.unwrap());
+            }
+        }
+    }
+}
+
+mod validated_optional_value_of {
```
Comment (mattforni) [inline src/tst/mod.rs]: Same re: name change

**Comment by mattforni** [comment]
Looks pretty solid to me, though I would like the name of that function changed. Also, please bump the minor version and reset patch to `0`. Once you do that, I'll merge and build out the new version of the package

File: src/lib.rs [hunk 4]
```diff
@@ -288,15 +288,44 @@ impl Args {
...
+
     /// Retrieves the value of the `Opt` identified by `opt_name`, casts it to
     /// the type specified by `T` and then runs all provided `Validation`s.
     ///
     /// # Failures
     ///
     /// Returns `Err(ArgsError)` if no `Opt` correspond to `opt_name`, if the value cannot
     /// be cast to type `T` or if any validation is considered invalid.
-    pub fn validated_value_of<T: FromStr>(
-            &self, opt_name: &str, validations: &[Box<Validation<T=T>>]) -> Result<T, ArgsError> {
```
Comment (philgebhardt) [inline src/lib.rs:299]: Just formatting change, using `where T: ...`

**Comment by philgebhardt** [comment]
- renamed functions (and made them public where necessary)
- bumped minor version -> `2.1.0`

**Pull request merged**